src/module_2/groceries_analysis.md

```diff
@@ -0,0 +1,1811 @@
+
+
+
+## Check Users
+
+Let's know try to combine the regulars dataframe with the users dataframe, to understand better the consumers behavior.
+
+
+```python
+# Calculate number of regulars by user
+regulars_by_user = regulars_df.groupby('user_id')['variant_id'].nunique().reset_index().rename(columns={'variant_id': 'regulars_count'})
+# Merge with users to get user information
+users = pd.merge( users_df,regulars_by_user, on='user_id', how='left').fillna({'regulars_count': 0})
+users.describe()
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>count_people</th>
+      <th>count_adults</th>
+      <th>count_children</th>
+      <th>count_babies</th>
+      <th>count_pets</th>
+      <th>regulars_count</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>count</th>
+      <td>325.000000</td>
+      <td>325.000000</td>
+      <td>325.000000</td>
+      <td>325.000000</td>
+      <td>325.000000</td>
+      <td>4983.000000</td>
+    </tr>
+    <tr>
+      <th>mean</th>
+      <td>2.787692</td>
+      <td>2.003077</td>
+      <td>0.707692</td>
+      <td>0.076923</td>
+      <td>0.636923</td>
+      <td>2.481437</td>
+    </tr>
+    <tr>
+      <th>std</th>
+      <td>1.365753</td>
+      <td>0.869577</td>
+      <td>1.026246</td>
+      <td>0.289086</td>
+      <td>0.995603</td>
+      <td>8.890588</td>
+    </tr>
+    <tr>
+      <th>min</th>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+    </tr>
+    <tr>
+      <th>25%</th>
+      <td>2.000000</td>
+      <td>2.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+    </tr>
+    <tr>
+      <th>50%</th>
+      <td>3.000000</td>
+      <td>2.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+    </tr>
+    <tr>
+      <th>75%</th>
+      <td>4.000000</td>
+      <td>2.000000</td>
+      <td>1.000000</td>
+      <td>0.000000</td>
+      <td>1.000000</td>
+      <td>1.000000</td>
+    </tr>
+    <tr>
+      <th>max</th>
+      <td>8.000000</td>
+      <td>7.000000</td>
+      <td>6.000000</td>
+      <td>2.000000</td>
+      <td>6.000000</td>
+      <td>320.000000</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Most people don't have a regular list, which is expected, as it's most likely voluntary info. Let's check the number of regulars by user_segment
+
+
+```python
+# distribution of user_segment values
+users['user_segment'].value_counts().plot.pie(y='user_segment',autopct='%1.1f%%')
```

Comment:
I would discourage you to use pie plots in general. Humans are not good at comparing angles. They could be arguable ok in some presentations with two or three elements where they look cool and there is huge differnce between the classes.